File: melosys-eessi-app/src/test/java/no/nav/melosys/eessi/service/sed/SedDataStub.java
package no.nav.melosys.eessi.service.sed;

import java.io.IOException;
import java.net.URI;
import java.net.URISyntaxException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.Objects;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
import no.nav.melosys.eessi.controller.dto.SedDataDto;

public class SedDataStub {

    public static SedDataDto getStub() throws IOException, URISyntaxException {
        URI søknadURI = Objects.requireNonNull(SedDataStub.class.getClassLoader().getResource("mock/sedDataDtoStub.json")).toURI();
        String json = Files.readString(Paths.get(søknadURI));
        var objectMapper = new ObjectMapper();
        objectMapper.registerModule(new JavaTimeModule());
        return objectMapper.readValue(json, SedDataDto.class);
    }
}

File: melosys-eessi-app/src/test/java/no/nav/melosys/eessi/service/sed/SedDataStub.kt
package no.nav.melosys.eessi.service.sed

import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import com.fasterxml.jackson.module.kotlin.readValue
import no.nav.melosys.eessi.controller.dto.SedDataDto
import no.nav.melosys.eessi.models.sed.SED
import no.nav.melosys.eessi.service.sed.mapper.til_sed.SedMapper
import java.nio.file.Files
import java.nio.file.Paths

object SedDataStub {
    @JvmStatic
    fun getStub(): SedDataDto = getStub("mock/sedDataDtoStub.json")

    fun getStub(fileName: String): SedDataDto {
        val søknadURI =
            SedDataStub::class.java.classLoader.getResource(fileName) ?: throw RuntimeException("Fant ikke filen $fileName")
        val json = Files.readString(Paths.get(søknadURI.toURI()))
        val objectMapper = jacksonObjectMapper().apply {
            registerModule(JavaTimeModule())
        }
        return objectMapper.readValue<SedDataDto>(json)
    }

    inline fun <reified T : SedMapper> mapTilSed(
        erCDM4_3: Boolean,
        testData: String,
        noinline block: SedDataDto.() -> Unit = {}
    ): SED = T::class.java.getDeclaredConstructor().newInstance().mapTilSed(getStub(testData).apply {
        block()
    }, erCDM4_3)
}
